application/src/pages/Cart/Cart.tsx

```diff
@@ -1,55 +1,44 @@
-import React, { useEffect } from 'react';
-import { useTranslation } from 'react-i18next';
+
+import React from 'react';
 import { useSelector } from 'react-redux';
-import { NavLink, useNavigate } from 'react-router-dom';
 import { redirect } from '../../bll/redirect';
 import { CartPropTypes } from '../../bll/reducers/cartReducer';
 import { AppStateType } from '../../bll/store';
-import { PATH } from '../PageRoutes/PageRoutes';
-import { CarInCart } from './CarInCart';
+import { CarInCart } from './CarInCart/CarInCart';
 import style from './Cart.module.css';
 import { CartForm } from './CartForm/CartForm';
+import { CartFormHeader } from './CartForm/CartFormHeader/CartFormHeader';
+import { CartHeader } from './CartHeader/CartHeader';
+import { EmptyCart } from './EmptyCart/EmptyCart';
 
 export const Cart: React.FC = () => {
   const cartReducer = useSelector<AppStateType, CartPropTypes[]>(
     (state) => state.cartReducer
   );
-  const navigate = useNavigate();
-  const { t } = useTranslation();
-
-  useEffect(() => {
-    if (cartReducer.length === 0) {
-      navigate(PATH.CART);
-    }
-  }, [cartReducer]);
+  const paymentCarDReducerTotal = useSelector<AppStateType, number>(
+    (state) => state.paymentCardReducer.total
+  );
+  const orderNum = useSelector<AppStateType, string>(
+    (state) => state.paymentCardReducer.orderNum
+  );
   redirect();
-  if (cartReducer.length < 1) {
-    return (
-      <div className={style.mainBlock}>
-        <h1>{t('cart.cartIsEmpty')}</h1>
-        <h3>
-          <NavLink className={style.link} to={PATH.CARS}>
-            {t('messages.startShopping')}
-          </NavLink>
-        </h3>
-      </div>
-    );
-  }
 
   let cartTotalSum = 0;
   cartReducer.forEach((car) => {
     cartTotalSum += car.total;
   });
 
+  const totalCarPrice = Math.floor(cartTotalSum * 100) / 100;
+
+  if (cartReducer.length < 1) {
```

Comment:
It's hard to understand what is role for that condition, so it's better

```
const isCartEmpty = cartReducer.length < 1;

if (isCartEmpty) {
... your code here
}
```

Reply:
fixed